.github/copilot-instructions.md

```diff
@@ -0,0 +1,79 @@
+# GitHub Copilot Code Review Instructions
+
+When performing a code review, respond in English.
+
+## Architecture & Patterns
+
+When performing a code review, ensure ViewModels are never injected as dependencies into services or repositories. Only Android activities and composable functions should use ViewModels.
+
+When performing a code review, verify that all async operations use `viewModelScope.launch` instead of `GlobalScope.launch` in ViewModels.
+
+When performing a code review, ensure Repository pattern is followed with proper data source abstraction.
+
+When performing a code review, verify StateFlow is used for reactive state management in ViewModels with proper `_uiState` and `uiState` pattern.
+
+## Error Handling & Safety
+
+When performing a code review, flag any use of the not-null assertion operator (`!!`) and suggest safe calls (`?.`) with proper null handling.
+
+When performing a code review, ensure all coroutine operations use proper error handling with `runCatching`
+
+When performing a code review, verify that all Logger calls include the TAG as context parameter, e.g., `Logger.warn("message", e = e, context = TAG)`.
+
+When performing a code review, check that methods exist before they are called, especially after refactoring.
+
+## UI & Compose Best Practices
+
+When performing a code review, ensure all user-facing strings use `stringResource(R.string.*)` instead of hardcoded strings.
+
+When performing a code review, verify that expensive Compose computations are wrapped in `remember` blocks.
+
+When performing a code review, check that Material3 design guidelines are followed for UI components.
+
+When performing a code review, ensure proper state management patterns with `MutableStateFlow` and `StateFlow`.
+
+## Code Quality & Readability
+
+When performing a code review, focus on readability and avoid nested ternary operators.
```

Comment:
> When performing a code review, focus on readability and avoid nested ternary operators.

I like the idea to focus on readability, but we're not dealing with nested ternary in Kotlin because it's not possible. Maybe instead, ask it to return early and avoid nested if-else wherever possible.

Example:

```kt
// do this:
fun decideCodepath(byDependency: Boolean) {
  if (!byDependency) return callAlternative()
  return callCodePath()
}

// instead of:
fun decideCodepath(byDependency: Boolean) {
  if (byDependency) {
    callCodePath()
  } else {
    callAlternative()
  }
}
```